ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -288,9 +295,35 @@ public synchronized List<Attestation> getAttestations(
         .filter(filterForCommitteeIndex)
         .flatMap(MatchingDataAttestationGroup::stream)
         .map(ValidatableAttestation::getAttestation)
+        .filter(
+            attestation ->
+                attestation.requiresCommitteeBits() == requestRequiresAttestationWithCommitteeBits)
+        .limit(attestationsSchema.getMaxLength())
         .toList();
   }
 
+  private SchemaDefinitions getSchemaDefinitions(final Optional<UInt64> maybeSlot) {
+    if (maybeSlot.isPresent()) {
+      return spec.atSlot(maybeSlot.get()).getSchemaDefinitions();
+    } else if (recentChainData.getCurrentSlot().isPresent()) {
+      return spec.atSlot(recentChainData.getCurrentSlot().get()).getSchemaDefinitions();
+    } else {
+      return spec.getGenesisSchemaDefinitions();
+    }
+  }
+
+  private Predicate<Map.Entry<UInt64, Set<Bytes>>> getSlotFilter(final Optional<UInt64> maybeSlot) {
+    if (maybeSlot.isPresent()) {
+      return (entry) -> entry.getKey().equals(maybeSlot.get());
+    } else {
+      return (entry) ->
+          recentChainData
+              .getCurrentSlot()
+              .map(currentSlot -> entry.getKey().equals(currentSlot))
+              .orElse(true);
```

Comment:
this is actually a non trivial change, the only time getCurrentSlot will return empty is if recentChainData isn't initialised...
i'd like to know why we're thinking about this change i think. there's no test justification or anything.

Reply:
Ok so I put back the old slot filtering logic and left the current slot usage to determine if we must filter Electra attestations. To summarise, here is the implemented logic:
- If the slot is provided, we filter the attestation based on it otherwise we retrieve all attestations
AND
- If the slot is provided, we use it to determine the milestone otherwise we use the current slot and finally we fallback to genesis: This let us know if we reached Electra and hence should filter Electra attestations only